### DISCUSSION TITLE
Is it possible to use the sqlite as a vector database with this ?

### DISCUSSION THREAD

**Discussion by RocketRene** [zombiezen/go-sqlite#91] [Q&A] [answered]
I am new to go and sqlite, and I want to build a small project that uses LLMs, embeddings and so on. I would be very happy if I would not have to run a real vector db because that's an overkill for my use case.  Can I use https://github.com/asg017/sqlite-vss and this library in combination ? 
And is it possible to generate type safe code with https://github.com/sqlc-dev/sqlc ?

**Comment by zombiezen** [comment, marked as answer]
> I am new to go and sqlite, and I want to build a small project that uses LLMs, embeddings and so on. I would be very happy if I would not have to run a real vector db because that's an overkill for my use case. Can I use https://github.com/asg017/sqlite-vss and this library in combination ?

Not directly, sadly, since this library does not support dynamic loading C extensions. You might be able to use [`modernc.org/ccgo/v4`](https://gitlab.com/cznic/ccgo) to translate the extension into Go and then hook it up with the [virtual table API](https://pkg.go.dev/zombiezen.com/go/sqlite#VTable).

> And is it possible to generate type safe code with https://github.com/sqlc-dev/sqlc ?

Again, I'm not aware of an ability to do that directly right now. According to the [sqlc docs](https://docs.sqlc.dev/en/latest/reference/config.html#gen), the codegen only supports `database/sql`. I'm not sure how hard it would be for them to add support for this package's API, but I imagine it could be done as a different backend.

**Reply by RocketRene** [reply]
Thank you very much for taking the time to write this answer :smile:   :pray: 
Then I will probably not use this library for now .. :( It would just make everything more difficult for me to understand.
